FeedStoreChallenge/InMemoryCacheFeedStore.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  InMemoryCacheFeedStore.swift
+//  FeedStoreChallenge
+//
+//  Created by panditpakhurde on 22/05/20.
+//  Copyright © 2020 Essential Developer. All rights reserved.
+//
+
+import Foundation
+
+public class InMemoryCacheFeedStore: FeedStore {
+    final class Cache {
+        internal let feed: [LocalFeedImage]
+        internal let timestamp: Date
+        
+        init(feed: [LocalFeedImage], timestamp: Date) {
+            self.feed = feed
+            self.timestamp = timestamp
+        }
+    }
+    
+    private final class Key {
+        private let key: String
+        
+        init(_ key: String) {
+            self.key = key
+        }
+    }
+    
+    private var key: Key
+    private var memoryCache = NSCache<Key, Cache>()
+    
+    public init(key: String) {
+        self.key = Key(key)
+    }
+    
+    public func deleteCachedFeed(completion: @escaping DeletionCompletion) {
+        if let _ = memoryCache.object(forKey: key) {
+            memoryCache.removeObject(forKey: key)
+        }
+
```

Comment:
You don't need to check if the value exists:

```suggestion
        memoryCache.removeObject(forKey: key)
```

Reply:
Yeah, right.